text/0113-canonical-metadata.md

```diff
@@ -0,0 +1,625 @@
+                    }
+                }
+            }
+        },
+        "AggregatingExport": {
+            "type": "object",
+            "properties": {
+                "exportsList": { 
+                    "type": "array",
+                    "items": {
+                        "type": "object",
+                        "properties": {
+                            "type": {
+                                "type": "string",
+                                "enum": ["class", "function", "expression"]
+                            }
+                        }
+                    }
+                },
+                "moduleSpecifier": { "$ref": "#/definitions/ModuleReference" }
+            }
+        },
+        "ApiDecorator": {
+            "type": "object",
+            "properties": {
+                "type": { "const": "api" },
+                "location": { "$ref": "#/definitions/SourceLocation" }
+            }
+        },
+        "Class": {
+            "type": "object",
+            "properties": {
+                "name": { "type": "string" },
+                "extends": { "$ref": "#/definitions/Class" },
+                "properties": { 
+                    "type": "array",
+                    "items": { "$ref": "#/definitions/ClassProperty" },
+                    "default": []
+                },
+                "methods": {
+                    "type": "array",
+                    "items": { "$ref": "#/definitions/ClassMethod" },
+                    "default": []
+                }
+            }
+        },
+        "ClassMemberType": {
+            "type": "string",
+            "enum": [ "property", "method"]
+        },
+        "ClassMethod": {
+            "type" : "object",
+            "properties": {
+                "name": { "type": "string" },
+                "accessType": {
+                    "type": "string",
+                    "enum": ["public", "private", "static"]
+                },
+                "decorator": { 
+                    "anyOf": [
+                        {"$ref": "#/definitions/ApiDecorator" },
+                        {"$ref": "#/definitions/TrackDecorator" },
+                        {"$ref": "#/definitions/WireDecorator" }
+                    ]
+                },
+                "parameterNames": { 
+                    "type": "array",
+                    "items": {
+                        "types": "string"
+                    }
+                },
+                "returnType": { "type": "string" },
+                "returnValue": { "type": [ "number", "string", "boolean", "null", "object" ] }
+            }
+        },
+        "ClassProperty": {
+            "type" : "object",
+            "properties": {
+                "accessType": {
+                    "type": "string",
+                    "enum": ["public", "private", "static"]
+                },
+                "dataProperty": { "type": "boolean"},
```

Comment:
What is this property used for?

Reply:
It represents a property defined on an instance. Our documents generalize them as properties. So I derived this term on the lines of data descriptor and accessor descriptor concept. Let me know if another name would describe it better.
```js
class MyComponent extends LightningElement {
         @track
         trackedProperty; // this is a data property

         privateField; // this is a data property

         get anotherProperty() { // For this one, hasGetter will be true
         }
}
```